packages/plugin-figma-blade-coverage/src/main.ts

```diff
@@ -0,0 +1,429 @@
+    figma.notify('⚠️ Error in rendering coverage card. Please try again', { error: true });
+    figma.closePlugin();
+  }
+};
+
+const calculateCoverage = (node: SceneNode): CoverageMetrics | null => {
+  let bladeComponents = 0;
+  let bladeTextStyles = 0;
+  let bladeColorStyles = 0;
+  let nonBladeComponents = 0;
+  let nonBladeTextStyles = 0;
+  let nonBladeColorStyles = 0;
+  let totalLayers = 0;
+
+  try {
+    if (getParentNode(node)?.type === 'PAGE' && !MAIN_FRAME_NODES.includes(node.type)) {
+      // if there are non-frame nodes as direct children of a page, ignore them
+      return null;
+    }
+
+    traverseNode(
+      node,
+      (traversedNode) => {
+        if (!traversedNode.visible) {
+          return;
+        }
+
+        if (
+          traversedNode.type === 'INSTANCE' &&
+          (BLADE_COMPONENT_IDS.includes(
+            (traversedNode.mainComponent?.parent as ComponentSetNode)?.key ?? '',
+          ) ||
+            BLADE_COMPONENT_IDS.includes(traversedNode.mainComponent?.key ?? ''))
+        ) {
+          bladeComponents++;
+        } else if (traversedNode.type === 'INSTANCE') {
+          nonBladeComponents++;
+          highlightNonBladeNode(traversedNode);
+        } else if (traversedNode.type === 'TEXT') {
+          // check if the text is using Blade's text styles
+          let isMixedTextStyleOfBlade = false;
+          let traversedNodeTextStyleId = '';
+          let isMixedColorStyleOfBlade = false;
+          let traversedNodeColorStyleId = '';
+
+          /** the textSyleId can have figma.mixed. so in that case we need to go character by character
+           * and do getRangeTextStyleId(charIndex,charIndex+1) instead of textStyleId
+           * */
+          if (traversedNode?.textStyleId === figma.mixed) {
+            isMixedTextStyleOfBlade = traversedNode.characters
+              .split('')
+              .every((character, index) => {
+                if (/\s/.test(character)) {
+                  return true;
+                }
+                return BLADE_TEXT_STYLE_IDS.includes(
+                  (traversedNode.getRangeTextStyleId(index, index + 1) as string).split(',')[0],
+                );
+              });
+
+            if (isMixedTextStyleOfBlade) {
+              bladeTextStyles++;
+            } else {
+              nonBladeTextStyles++;
+              highlightNonBladeNode(traversedNode);
+            }
+          } else {
+            traversedNodeTextStyleId = traversedNode?.textStyleId?.split(',')[0];
+
+            if (BLADE_TEXT_STYLE_IDS.includes(traversedNodeTextStyleId ?? '')) {
+              bladeTextStyles++;
+            } else {
+              nonBladeTextStyles++;
+              highlightNonBladeNode(traversedNode);
+            }
+          }
+
+          /** the fillstyleId can have figma.mixed. so in that case we need to go character by character
+           * and do getRangeFillStyleId(4,5) instead of fillStyleId
+           * */
```

Comment:
nit: beautify these comments, getting a bit hard to read and looks ugly. 

```
/**
 * The textSyleId can have figma.mixed. so in that case we need to go character by character
 * and do getRangeTextStyleId(charIndex,charIndex+1) instead of textStyleId
 */
```

> ps: vscode automatically creates multiline comments nicely if you try /** and press enter

Reply:
done